pkg/pillar/cmd/zedagent/reportinfo.go

```diff
@@ -550,6 +550,62 @@ func PublishDeviceInfoToZedCloud(ctx *zedagentContext) {
 	}
 }
 
+// PublishAppInstMetaDataToZedCloud is called when an appInst reports its Metadata to EVE.
+// AppInst metadata is relayed to the controller to be processed further.
+func PublishAppInstMetaDataToZedCloud(ctx *zedagentContext, appInstID string, appInstMetadata *types.AppInstMetaData, iteration int) {
+	log.Functionf("PublishAppInstMetaDataToZedCloud: appInstID: %v, metadataType: %v",
+		appInstID, appInstMetadata.Type)
+	var ReportInfo = &info.ZInfoMsg{}
+
+	contentType := new(info.ZInfoTypes)
+	*contentType = info.ZInfoTypes_ZiAppInstMetaData
+	ReportInfo.Ztype = *contentType
+	ReportInfo.DevId = *proto.String(zcdevUUID.String())
+	ReportInfo.AtTimeStamp = ptypes.TimestampNow()
+
+	ReportAppInstMetaData := new(info.ZInfoAppInstMetaData)
+
+	ReportAppInstMetaData.Type = info.AppInstMetaDataType(appInstMetadata.Type)
+	ReportAppInstMetaData.Uuid = appInstID
+	ReportAppInstMetaData.Data = appInstMetadata.Data
+
+	ReportInfo.InfoContent = new(info.ZInfoMsg_Amdinfo)
+	if reportAppInstMetadata, ok := ReportInfo.GetInfoContent().(*info.ZInfoMsg_Amdinfo); ok {
+		reportAppInstMetadata.Amdinfo = ReportAppInstMetaData
+	}
+
+	log.Functionf("PublishAppInstMetaDataToZedCloud sending %v", ReportInfo)
+
+	data, err := proto.Marshal(ReportInfo)
+	if err != nil {
+		log.Fatal("PublishAppInstMetaDataToZedCloud proto marshaling error: ", err)
+	}
+	statusURL := zedcloud.URLPathString(serverNameAndPort, zedcloudCtx.V2API, devUUID, "info")
+
+	zedcloud.RemoveDeferred(zedcloudCtx, appInstID)
```

Comment:
Here you are removing any deferred App Info message and down below set it using the same UUID.

You need a separate key for these messages; I 
An example is
	deferKey := "attest:" + zcdevUUID.String()
	zedcloud.RemoveDeferred(zedcloudCtx, deferKey)
then
		zedcloud.SetDeferred(zedcloudCtx, deferKey, buf, size, attestURL,
			true)

Reply:
Changed. 
Used `deferKey := "attest:" + appInstID` as the key for deferred message.

Reply:
Needs to be unique; attest is merely an example of a string that was already used by you can't use the same string since then they are not unique any more

Reply:
But I see the code uses deferKey := "appInstMetadataInfo:" + appInstID which is fine.